go_final_project-main/internal/handlers/task.go

```diff
@@ -0,0 +1,213 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go_final_project/internal/models"
+	"go_final_project/internal/utils"
+	"net/http"
+	"strconv"
+	"time"
+
+	_ "modernc.org/sqlite"
+)
+
+// Task обрабатывает HTTP-запросы для выполнения CRUD-операций над задачами в приложении-планировщике.
+// Он поддерживает методы GET, POST, PUT и DELETE.
+//
+// GET:
+//   - Извлекает задачу по её ID из базы данных.
+//   - Возвращает JSON-объект со всеми полями задачи.
+//   - Если ID пуст или не найден, возвращает ошибку 400 Bad Request.
+//
+// POST:
+//   - Создает новую задачу в базе данных.
+//   - Возвращает JSON-объект с ID созданной задачи.
+//   - Если тело запроса недействительно или отсутствуют поля, возвращает ошибку 400 Bad Request.
+//
+// PUT:
+//   - Обновляет существующую задачу в базе данных.
+//   - Возвращает пустой JSON-объект.
+//   - Если тело запроса недействительно или отсутствуют поля, возвращает ошибку 400 Bad Request.
+//   - Если указанный ID не найден в базе данных, возвращает ошибку 400 Bad Request.
+//
+// DELETE:
+//   - Удаляет задачу из базы данных по её ID.
+//   - Возвращает пустой JSON-объект.
+//   - Если ID пуст или не найден, возвращает ошибку 400 Bad Request.
+//
+// По умолчанию, если метод запроса не GET, POST, PUT или DELETE, возвращает ошибку 500 Internal Server Error.
+func (c *DBConnection) Task(w http.ResponseWriter, r *http.Request) {
+	c.Mu.Lock()
+	defer c.Mu.Unlock()
+	switch r.Method {
+	// В нашем случае необходимо добавить обработку GET-запроса, который возвратит все параметры задачи по её идентификатору.
+	// Если сейчас нажать на иконку редактирования задачи, появится ошибка.
+	// Исправьте ситуацию — реализуйте обработчик GET-запроса /api/task?id=<идентификатор>.
+	// Запрос должен возвращать JSON-объект со всеми полями задачи.
+	case http.MethodGet:
+		id := r.FormValue("id")
+		if id == "" {
+			utils.SendErr(w, errors.New("id is empty"), http.StatusBadRequest)
+			return
+		}
+		task := models.Task{}
+		rows, err := c.DB.Query(`SELECT id, date, title, comment, repeat FROM scheduler WHERE id = ?`, id)
```

Comment:
Так как нам нужна одна запись, то нет смысла запрашивать много строк и читать из них одну
```suggestion
		err := c.DB.QueryRow(`SELECT id, date, title, comment, repeat FROM scheduler WHERE id = ?`, id).Scan(.....)
```